### PR TITLE
[updategraph] After system upgrade, restore files/directories with original attributes etc.

### PR DESCRIPTION
**Pull request by zhenggen-xu**
[updategraph] After system upgrade, restore files/directories with original attributes etc.
Restore a few more files that was missed before.
Restore FRR configuration directory if exists on old system

Signed-off-by: Zhenggen Xu <zxu@linkedin.com>

This depends on https://github.com/Azure/sonic-utilities/pull/410

**- What I did**
Restore files/directories with original attributes.
Restore some missed configurations.

**- How I did it**
added missing files.  use "cp -ar"

**- How to verify it**
build image with frr, and after "sonic_installer" and reboot into the new image,  frr is restored with old attributes. 

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/image_config/updategraph/updategraph [hunk 1]
```diff
@@ -50,9 +50,9 @@ fi
 . /etc/sonic/updategraph.conf
 
 check_system_warm_boot
-
+copy_list="minigraph.xml snmp.yml acl.json config_db.json deployment_id_asn_map.yml frr"
```
Comment (taoyl-ms) [inline files/image_config/updategraph/updategraph]: I think `deployment_id_asn_map.yml` should come with image?

**Comment by zhenggen-xu** [comment]
@yxieca @taoyl-ms It is not very clear to me the usage of this file: deployment_id_asn_map.yml.  My understanding is this is used to set the remote-as in bgp configuration. For any upgrade,  we should keep all the configurations, including bgp configurations, I don't see why we should change the map during the upgrade even if new images have the new map.

For new switches builds, we could use the new map from new images.  Also, this file should be changed for different environment, it only gives default mapping looks like.

Let's me know your opinion, it is easy to remove that file from copy list but would like understand the reason.

**Comment by yxieca** [comment]
@zhenggen-xu 

I did 'git grep' in sonic-buildiamge, sonic-utilities, sonic-platform-common, sonic-quagga, sonic-frr, I didn't see anyone updating this file except in sonic-buildimage, it is used as an initial template.

It might be reasonable to move this file under /usr/share/sonic/template instead (in a different PR).

Unless I missed something in sub modules that I didn't check, the usage of this file is very limited and should not be copied forward. Whatever effect this file has in the old image has been captured in config_db.

Please let me know your opinion.

Regards,
Ying

**Comment by zhenggen-xu** [comment]
@yxieca It is not necessarily updated by SONiC code. I would think this file should be downloaded as part of initial configuration (like ZTP).  But during the upgrade process, I still think we should keep the old file so we have the same configuration.  This file is used in conjunction with config_db data to generate the bgp configuration:
sonic-cfggen -d -y /etc/sonic/deployment_id_asn_map.yml -t /usr/share/sonic/templates/bgpd.conf.j2 > /etc/quagga/bgpd.conf

**Comment by zhenggen-xu** [comment]
@lguohan  Can you comment on this PR so we can close the discussion?

**Comment by yxieca** [comment]
Made to 201811 branch on 2/2/2019

**Pull request merged**